TempMail/ContentView.swift

```diff
@@ -8,49 +8,110 @@
 
 import SwiftUI
 
+
 struct ContentView: View {
+    @State var orientation: UIDeviceOrientation = UIDevice.current.orientation
 //    variable for Email struct
     @ObservedObject var email: Email = Email()
     //this variable is for later, when I implement a feature that will save user data even when the app is closed
     private let defaults = UserDefaults.standard
     
+    // TODO: can the header be abstracted? lots of repeated code...
     var body: some View {
-        VStack {
-//          title
-            Text("Temp Mail")
-                .font(.largeTitle)
-                .padding()
-            Spacer()
-//          Generate new email button
-            Button(action: {
-                self.email.setNewEmailAddr()
-            }) {
-                Text("Generate New Email")
-                    .fontWeight(.bold)
-                    .foregroundColor(.blue)
-                    .padding()
-                    .overlay(RoundedRectangle(cornerRadius: 40)
-                        .stroke(Color.blue, lineWidth: 5))
-            }
-            
-            Spacer()
-//          display email address for user
-            Text("Current Address: \(self.email.getEmailAddr())@1secmail.com")
-                .padding()
-            
-            Spacer()
-//          List of emails attached to above address
-            List {
-                MessageRow(messageInfo: InboxModel(id: 1234,
-                                                   from: "sender@example.com",
-                                                   subject: "example subject",
-                                                   date: "Today"))
-                MessageRow(messageInfo: InboxModel(id: 1234,
-                                                   from: "sender@example.com",
-                                                   subject: "example subject",
-                                                   date: "Today"))
+        return HStack {
+            if orientation.isPortrait || orientation.isFlat { // TODO: test more to ensure that this is desired behavior
```

Comment:
I believe if you removed the orientation.isFlat part it would fix the bug we were talking about earlier through text